zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ElasticsearchBatchRequest.java

```diff
@@ -0,0 +1,268 @@
+import co.elastic.clients.elasticsearch.core.bulk.BulkResponseItem;
+import co.elastic.clients.elasticsearch.core.bulk.UpdateAction;
+import co.elastic.clients.elasticsearch.core.bulk.UpdateOperation;
+import io.camunda.exporter.entities.ExporterEntity;
+import io.camunda.exporter.exceptions.PersistenceException;
+import io.camunda.exporter.utils.ElasticsearchScriptBuilder;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ElasticsearchBatchRequest implements BatchRequest {
+  public static final int UPDATE_RETRY_COUNT = 3;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchBatchRequest.class);
+  private final ElasticsearchClient esClient;
+  private final BulkRequest.Builder bulkRequestBuilder;
+  private final ElasticsearchScriptBuilder scriptBuilder;
+
+  public ElasticsearchBatchRequest(
+      final ElasticsearchClient esClient,
+      final BulkRequest.Builder bulkRequestBuilder,
+      final ElasticsearchScriptBuilder scriptBuilder) {
+    this.esClient = esClient;
+    this.bulkRequestBuilder = bulkRequestBuilder;
+    this.scriptBuilder = scriptBuilder;
+  }
+
+  @Override
+  public BatchRequest add(final String index, final ExporterEntity entity)
+      throws PersistenceException {
+    return addWithId(index, entity.getId(), entity);
+  }
+
+  @Override
+  public BatchRequest addWithId(final String index, final String id, final ExporterEntity entity)
+      throws PersistenceException {
+    LOGGER.debug("Add index request for index {} id {} and entity {} ", index, id, entity);
+    final var idx = indexBuilder(index, id, entity).build();
+    final var op = bulkOperationBuilder(idx).build();
+    bulkRequestBuilder.operations(op);
+    return this;
+  }
+
+  @Override
+  public BatchRequest addWithRouting(
+      final String index, final ExporterEntity entity, final String routing)
+      throws PersistenceException {
+    LOGGER.debug(
+        "Add index request with routing {} for index {} and entity {} ", routing, index, entity);
+    final var idx = indexWithRoutingBuilder(index, entity, routing).build();
+    final var op = bulkOperationBuilder(idx).build();
+    bulkRequestBuilder.operations(op);
+    return this;
+  }
+
+  @Override
+  public BatchRequest upsert(
+      final String index,
+      final String id,
+      final ExporterEntity entity,
+      final Map<String, Object> updateFields)
+      throws PersistenceException {
+    LOGGER.debug(
+        "Add upsert request for index {} id {} entity {} and update fields {}",
+        index,
+        id,
+        entity,
+        updateFields);
+    final var upsert = upsertWithDocBuilder(entity, updateFields).build();
+    final var updateOperation = updateOperationBuilder(index, id, upsert).build();
+    final var op = bulkOperationBuilder(updateOperation).build();
+    bulkRequestBuilder.operations(op);
+
+    return this;
```

Comment:
:wrench: : This can also be replaced with 
```
return upsertWithRouting(index, id, entity, updateFields, null);
```

According the the ES interface routing is nullable. Might be also applicable to other methods that has both variants with and without routing.